pyvista/plotting/scalar_bars.py

```diff
@@ -28,12 +28,34 @@ def __repr__(self):
         """Nice representation of this class."""
         lines = []
         lines.append('Scalar Bar Title     Interactive')
-        for title, actor in self._scalar_bar_actors.items():
+        for title in self._scalar_bar_actors:
             interactive = title in self._scalar_bar_widgets
             title = f'"{title}"'
             lines.append(f'{title:20} {str(interactive):5}')
         return '\n'.join(lines)
 
+    def _remove_mapper_from_plotter(self, actor, reset_camera=False, render=False):
+        """Remove an actor's mapper from the given plotter's _scalar_bar_mappers."""
+        try:
+            mapper = actor.GetMapper()
+        except AttributeError:
+            return
+        for name in self._scalar_bar_mappers:
```

Comment:
```suggestion
        for name in list(self._scalar_bar_mappers):
```

Reply:
Implementing this since you're right, dictionary keys would change during the loop.  Should be triggering an error in CI/CD, so I'm not sure what the issue is there.